modules/renter/upload.go

```diff
@@ -74,6 +70,15 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 			up.PieceSize = smallPieceSize
 		}
 	}
+	// calculate end height
+	// TODO: using 0 as a special value is kind of hacky. Should probably have
+	// an explicit "renew" field.
+	var endHeight types.BlockHeight
+	if up.Duration == 0 {
+		endHeight = 0
+	} else {
+		endHeight = r.cs.Height() + up.Duration
+	}
```

Comment:
Ah, okay. For a while I was thinking that all files should just be renewed indefinitely, but I can see use cases where you'd want to automatically delete files after a number of days (example: CCTV footage)

Setting `duration=0` is definitely industry-standard for indicating infinity, but I think that it would cause edge cases, so I agree that an alternate approach should be used.